src/simulation/elements/TSNS.cpp

```diff
@@ -82,7 +84,35 @@ int Element_TSNS::update(UPDATE_FUNC_ARGS)
 					continue;
 				if ((r&0xFF)!=PT_TSNS && (r&0xFF)!=PT_METL && parts[r>>8].temp > parts[i].temp)
 					parts[i].life = 1;
+				if ((r & 0xFF) != PT_FILT && ((r&0xFF) != PT_TSNS && parts[i].tmp == 1))
+				{
+					setFilt = true;
+					photonWl = parts[r >> 8].temp;
+				}
 			}
+	if (setFilt)
+	{
+		int nx, ny;
+		for (rx = -1; rx<2; rx++)
+			for (ry = -1; ry<2; ry++)
+				if (BOUNDS_CHECK && (rx || ry))
+				{
+					r = pmap[y + ry][x + rx];
+					if (!r)
+						continue;
+					nx = x + rx;
+					ny = y + ry;
+					while ((r & 0xFF) == PT_FILT)
+					{
+						parts[r >> 8].ctype = 0x10000000+ceil(photonWl);
+						nx += rx;
+						ny += ry;
+						if (nx<0 || ny<0 || nx >= XRES || ny >= YRES)
+							break;
+						r = pmap[ny][nx];
```

Comment:
What is the point of lines 108-112 here?

Reply:
The point of those lines is to copy the value in a line vertical, diagonal, or horizontal just like DTEC does. if it was an if statement, it wouldn't work correctly.